net/graphsync_fetcher.go

```diff
@@ -236,9 +239,10 @@ func (gsf *GraphSyncFetcher) fetchRemainingTipsets(ctx context.Context, starting
 func (gsf *GraphSyncFetcher) fullBlockSel() ipld.Node {
 	selector := gsf.ssb.ExploreFields(func(efsb selectorbuilder.ExploreFieldsSpecBuilder) {
 		efsb.Insert("messages", gsf.ssb.ExploreFields(func(messagesSelector selectorbuilder.ExploreFieldsSpecBuilder) {
-			messagesSelector.Insert("secpRoot", gsf.ssb.Matcher())
+			messagesSelector.Insert("secpRoot", gsf.fetchThroughAMTSelector(10))
+			messagesSelector.Insert("bLSRoot", gsf.fetchThroughAMTSelector(10))
 		}))
-		efsb.Insert("messageReceipts", gsf.ssb.Matcher())
+		efsb.Insert("messageReceipts", gsf.fetchThroughAMTSelector(10))
```

Comment:
Please factor out this `10`. I think the intention is that this fetches "everything" so either leave an extensive comment justifying the value 10, or set this to something like 1000 so this doesn't cause a very hard to diagnose failure later.

Reply:
Recursion depth of 10 allows us to retrieve a amt with a billion elements. Recursion depth of 1000 makes my head hurt. Arguably it should be much lower. I extracted it to a constant and documented the choice.